x-pack/plugins/monitoring/server/config.test.ts

```diff
@@ -0,0 +1,157 @@
+          "api_polling_frequency": "PT30S",
+        },
+        "tests": Object {
+          "cloud_detector": Object {
+            "enabled": true,
+          },
+        },
+        "ui": Object {
+          "ccs": Object {
+            "enabled": true,
+          },
+          "container": Object {
+            "elasticsearch": Object {
+              "enabled": false,
+            },
+            "logstash": Object {
+              "enabled": false,
+            },
+          },
+          "elasticsearch": Object {
+            "apiVersion": "master",
+            "customHeaders": Object {},
+            "healthCheck": Object {
+              "delay": "PT2.5S",
+            },
+            "ignoreVersionMismatch": false,
+            "logFetchCount": 10,
+            "logQueries": false,
+            "pingTimeout": "PT30S",
+            "preserveHost": true,
+            "requestHeadersWhitelist": Array [
+              "authorization",
+            ],
+            "requestTimeout": "PT30S",
+            "shardTimeout": "PT30S",
+            "sniffInterval": false,
+            "sniffOnConnectionFault": false,
+            "sniffOnStart": false,
+            "ssl": Object {
+              "alwaysPresentCertificate": false,
+              "keystore": Object {},
+              "truststore": Object {},
+              "verificationMode": "full",
+            },
+            "startupTimeout": "PT5S",
+          },
+          "enabled": true,
+          "logs": Object {
+            "index": "filebeat-*",
+          },
+          "max_bucket_size": 10000,
+          "min_interval_seconds": 10,
+          "show_license_expiration": true,
+        },
+      }
+    `);
+  });
+});
+
+describe('createConfig()', () => {
+  it('should wrap in Elasticsearch config', async () => {
+    const config = createConfig(
+      configSchema.validate({
+        elasticsearch: {
+          hosts: 'http://localhost:9200',
+        },
+        ui: {
+          elasticsearch: {
+            hosts: 'http://localhost:9200',
+          },
+        },
+      })
+    );
+    expect(config.elasticsearch.hosts).toEqual(['http://localhost:9200']);
+    expect(config.ui.elasticsearch.hosts).toEqual(['http://localhost:9200']);
+  });
+
+  it('should attempt to read ssl files', async () => {
+    let error;
+    try {
+      createConfig(
+        configSchema.validate({
+          ui: {
+            elasticsearch: {
+              hosts: 'http://localhost:9200',
+              ssl: {
+                certificate: 'packages/kbn-dev-utils/certs/elasticsearch.crt',
+                key: 'packages/kbn-dev-utils/certs/elasticsearch.key',
+                certificateAuthorities: 'packages/kbn-dev-utils/certs/ca.crt',
+                verificationMode: 'full',
+              },
+            },
+          },
+        })
+      );
+    } catch (err) {
+      error = err.message;
+    }
+    expect(error.includes('no such file or directory')).toBe(true);
+  });
```

Comment:
... you can change this test to be a bit more accurate.

```suggestion
  it('should attempt to read PEM files', async () => {
    const ssl = {
      certificate: 'packages/kbn-dev-utils/certs/elasticsearch.crt',
      key: 'packages/kbn-dev-utils/certs/elasticsearch.key',
      certificateAuthorities: 'packages/kbn-dev-utils/certs/ca.crt',
    };
    const config = createConfig(
      configSchema.validate({
        elasticsearch: {
          ssl,
        },
        ui: {
          elasticsearch: {
            ssl,
          },
        },
      })
    );

    const expected = expect.objectContaining({
      certificate: 'contents-of-packages/kbn-dev-utils/certs/elasticsearch.crt',
      key: 'contents-of-packages/kbn-dev-utils/certs/elasticsearch.key',
      certificateAuthorities: ['contents-of-packages/kbn-dev-utils/certs/ca.crt'],
    });
    expect(config.elasticsearch.ssl).toEqual(expected);
    expect(config.ui.elasticsearch.ssl).toEqual(expected);
  });
```

Normally I'm not a fan of depending on external module behavior in unit tests, but I think it makes sense here, given what we want to verify :)

Reply:
Great suggestion!